central/sensor/service/pipeline/clustermetrics/update_maxima.go

```diff
@@ -0,0 +1,41 @@
+package clustermetrics
+
+import (
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/maputil"
+	"github.com/stackrox/rox/pkg/set"
+)
+
+// BillingMetrics are the metrics we collect and show to the customers to help
+// them report their usage.
+type BillingMetrics struct {
+	TotalNodes int64
+	TotalCores int64
+}
+
+var (
+	nodesMap = maputil.NewMaxMap[string, int64]()
```

Comment:
I'm wondering if instead of maintaining such a max cache, we could instead upsert the value directly on each measurement point (with e.g. timestamp rounded to full hours). Then Central would not have to maintain any state at the cost of some additional database queries.

Reply:
Right, this is a reasonable option. It would definitely increase the traffic to the DB which is probably fine.

In any case, for consistency and locality reasons, code from this file should move to the datastore. If you want to keep a cache to avoid frequent upserts, place it into the related datastore.